src/Composer/Installer/ExtensionInstaller.php

```
@@ -0,0 +1,163 @@
+     * @param string      $type
+     * @param Filesystem  $filesystem
+     */
+    public function __construct(IOInterface $io, Composer $composer, $type = 'extension', Filesystem $filesystem = null)
+    {
+        $this->composer = $composer;
+        $this->downloadManager = $composer->getDownloadManager();
+        $this->io = $io;
+        $this->type = $type;
+
+        $this->filesystem = $filesystem ?: new Filesystem();
+        $this->vendorDir = rtrim($composer->getConfig()->get('vendor-dir'), '/');
+        $this->binDir = rtrim($composer->getConfig()->get('bin-dir'), '/');
+        $this->pickle = "/home/pierre/projects/pickle/pickle/bin/pickle";
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    public function supports($packageType)
+    {
+        return $packageType === $this->type || null === $this->type;
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    public function isInstalled(InstalledRepositoryInterface $repo, PackageInterface $package)
+    {
+		return false;
+        //return $repo->hasPackage($package) && is_readable($this->getInstallPath($package));
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    public function install(InstalledRepositoryInterface $repo, PackageInterface $package)
+    {		
+		$this->io->write("Pickle: fetching " . $package->getName());
+
+		$dist_url = $package->getDistURL();
+		if (strtolower(substr($dist_url, -3)) == "zip") {
+			$extract_dir = $this->uncompress($package);
+			$json = $this->findComposerJson($extract_dir);
+		} else {
+			$pkg_dir = $this->createTempDir();
+			$this->downloadManager->download($package, $pkg_dir);
+		}
+		$process = new ProcessExecutor($this->io);
+		/* Add interactions */
+		$cmd = sprintf('%s install -q -n --save-logs=%s %s', $this->pickle, ProcessExecutor::escape($pkg_dir . 'logs'), ProcessExecutor::escape($pkg_dir));
+		$process->execute($cmd);
+		return;
+    }
+
+	protected function createTempDir()
+	{
+		$tmpdir = sys_get_temp_dir();
+		if (!$tmpdir) {
```

you should try creating a temp folder inside the composer cache in case the system temp dir cannot be used, before failing hard